zhmc_prometheus_exporter/zhmc_prometheus_exporter.py

```diff
@@ -2187,18 +2199,85 @@ def main():
         logprint(logging.INFO, PRINT_ALWAYS,
                  f"Verbosity level: {VERBOSE_LEVEL}")
 
-        hmccreds_filename = args.c
+        config_filename = args.c
         logprint(logging.INFO, PRINT_V,
-                 f"Parsing HMC credentials file: {hmccreds_filename}")
-        yaml_creds_content = parse_yaml_file(
-            hmccreds_filename, 'HMC credentials file', 'hmccreds_schema.yaml')
-        # metrics is required in the metrics schema:
-        yaml_creds = yaml_creds_content["metrics"]
+                 f"Parsing exporter config file: {config_filename}")
+        config_dict = parse_yaml_file(
+            config_filename, 'exporter config file', 'config_schema.yaml')
+
+        if 'metrics' not in config_dict and 'hmc' not in config_dict:
+            new_exc = ImproperExit(
+                "The exporter config file must specify either the new 'hmc' "
+                "item or the old 'metrics' item, but it specifies none.")
+            new_exc.__cause__ = None  # pylint: disable=invalid-name
+            raise new_exc
+
+        if 'metrics' in config_dict and 'hmc' in config_dict:
+            new_exc = ImproperExit(
+                "The exporter config file must specify either the new 'hmc' "
+                "item or the old 'metrics' item, but it specifies both.")
+            new_exc.__cause__ = None  # pylint: disable=invalid-name
+            raise new_exc
+
+        if 'version' in config_dict and config_dict['version'] != 2:
+            new_exc = ImproperExit(
+                "The exporter config file must have the version 2 format, "
+                "but it specifies the version {} format.".
+                format(config_dict['version']))
+            new_exc.__cause__ = None  # pylint: disable=invalid-name
+            raise new_exc
+
+        if 'metrics' in config_dict and 'hmc' not in config_dict:
+            logprint(logging.WARNING, PRINT_ALWAYS,
+                     "DEPRECATED: The exporter config file has the old version "
+                     "1 format. Please migrate to the new version 2 format "
+                     "(use --help-config). The version 1 format is deprecated "
+                     "and support for it will be removed in a future release.")
+
+            # Convert old format to new format
```

Comment:
I am wondering if it would be worth to turn this into a helper method that the user can invoke, too.
Meaning: would it make sense to include a little "format conventer" feature here that users can call to update their files? But then, it would probably be a bit more work to get to a good converter that is robust enough.

Reply:
If we do this, it would need to be a command the user can invoke, so they don't have to program. After all, this is a one time migration that is described in the docs and can be done in 5 minutes (including the reading how to do it).

I don't want to provide a separate command, so it would be an option on the zhmc_prometheus_exporter command which then performs just the migration (i.e. not in addition to running the exporter).

Reply:
Sounds good to me. 
I do think a "breaking" format version change is somehow different from a "compatible" change. And I think it is not something that we have to release immediately, right.